nipype/interfaces/freesurfer/petsurfer.py

```diff
@@ -0,0 +1,594 @@
+
+    no_pvc = traits.Bool(
+        argstr="--no-pvc",
+        desc="turns off PVC entirely (both PSF and TFE)",
+    )
+
+    segpvfres = traits.Float(
+        argstr="--segpvfres %f",
+        desc="set the tissue fraction resolution parameter (def is 0.5)",
+    )
+
+    rbv = traits.Bool(argstr="--rbv", requires=["subjects_dir"], desc="perform RBV PVC")
+
+    rbv_res = traits.Float(
+        argstr="--rbv-res %f",
+        desc="voxsize : set RBV voxel resolution (good for when standard res takes too much memory)",
+    )
+
+    mg = traits.List(argstr="--id %s...", desc="Manually specify segmentation ids")
+
+    mg_ref_cerebral_wm = traits.Bool(
+        argstr="--mg-ref-cerebral-wm", desc=" set MG RefIds to 2 and 41"
+    )
+
+    mg_ref_lobes_wm = traits.Bool(
+        argstr="--mg-ref-lobes-wm",
+        desc="set MG RefIds to those for lobes when using wm subseg",
+    )
+
+    mgx = traits.Float(
+        argstr="--mgx %f",
+        desc="gmxthresh : GLM-based Mueller-Gaertner PVC, gmxthresh is min gm pvf bet 0 and 1",
+    )
+
+    km_ref = traits.List(
+        argstr="--km-ref %s...",
+        desc="RefId1 RefId2 ... : compute reference TAC for KM as mean of given RefIds",
+    )
+
+    km_hb = traits.List(
+        argstr="--km-hb %s...",
+        desc="RefId1 RefId2 ... : compute HiBinding TAC for KM as mean of given RefIds",
+    )
+
+    ss = traits.Tuple(
+        traits.Float,
+        traits.Float,
+        traits.Float,
+        argstr="--ss %f %f %f",
+        desc="bpc scale dcf : steady-state analysis spec blood plasma concentration, unit scale and decay correction factor. You must also spec --km-ref. Turns off rescaling",
+    )
+
+    X = traits.Bool(
+        argstr="--X", desc="save X matrix in matlab4 format as X.mat (it will be big)"
+    )
+
+    y = traits.Bool(argstr="--y", desc="save y matrix in matlab4 format as y.mat")
+
+    beta = traits.Bool(
+        argstr="--beta", desc="save beta matrix in matlab4 format as beta.mat"
+    )
+
+    X0 = traits.Bool(
+        argstr="--X0",
+        desc="save X0 matrix in matlab4 format as X0.mat (it will be big)",
+    )
+
+    save_input = traits.Bool(
+        argstr="--save-input", desc="saves rescaled input as input.rescaled.nii.gz"
+    )
+
+    save_eres = traits.Bool(argstr="--save-eres", desc="saves residual error")
+
+    save_yhat = traits.Bool(
+        argstr="--save-yhat",
+        xor=["save_yhat_with_noise"],
+        desc="save signal estimate (yhat) smoothed with the PSF",
+    )
+
+    save_yhat_with_noise = traits.Tuple(
+        traits.Int,
+        traits.Int,
+        argstr="--save-yhat-with-noise %i %i",
+        xor=["save_yhat"],
+        desc="seed nreps : save signal estimate (yhat) with noise",
+    )
+
+    save_yhat_full_fov = traits.Bool(
+        argstr="--save-yhat-full-fov", desc="save signal estimate (yhat)"
+    )
+
+    save_yhat0 = traits.Bool(argstr="--save-yhat0", desc="save signal estimate (yhat)")
+
+    opt = traits.Int(
```

Comment:
```suggestion
    optimization_schema = traits.Int(
```

And does it make sense to do an enum and then map?

```suggestion
    optimization_schema = traits.Enum(
        "3D", "2D", "1D", "3D_MB", "2D_MB", "1D_MB", "MBZ", "MB3",
```

Below you would need to convert to an integer before formatting:

```Python
    def _format_arg(self, opt, spec, val):
        if opt == "optimization_schema":
            val = ["3D", "2D", "1D", "3D_MB", "2D_MB", "1D_MB", "MBZ", "MB3"].index(val) + 1
        super()._format_arg(opt, spec, val)

Reply:
So you mean that the input should be defined as 

..inputs.optimization_schema = "3D_MB"

and then the command line should be "--opt 3" (for position 4)?

I can't seem to get this to work when doing what you are suggesting,

Reply:
I have updated to this, but still doesn't work as expected

`def _format_arg(self, optimization_schema, spec, val):
        if optimization_schema:
            val = ["3D", "2D", "1D", "3D_MB", "2D_MB", "1D_MB", "MBZ", "MB3"].index(optimization_schema)
            return val
        return super()._format_arg(optimization_schema, spec, val)`

Reply:
Solved with newest commit. Thanks @effigies!